src/python/gudhi/datasets/remote.py

```diff
@@ -0,0 +1,206 @@
+    """
+    data_home = get_data_home(data_home)
+    shutil.rmtree(data_home)
+
+def _checksum_sha256(file_path):
+    """
+    Compute the file checksum using sha256.
+
+    Parameters
+    ----------
+    file_path: string
+        Full path of the created file.
+
+    Returns
+    -------
+        The hex digest of file_path.
+    """
+    sha256_hash = hashlib.sha256()
+    chunk_size = 4096
+    with open(file_path,"rb") as f:
+        # Read and update hash string value in blocks of 4K
+        while True:
+            buffer = f.read(chunk_size)
+            if not buffer:
+                break
+            sha256_hash.update(buffer)
+    return sha256_hash.hexdigest()
+
+def _fetch_remote(url, filename, dirname, file_checksum = None, accept_license = False):
+    """
+    Fetch the wanted dataset from the given url and save it in file_path.
+
+    Parameters
+    ----------
+    url : string
+        The url to fetch the dataset from.
+    filename : string
+        The name to give to downloaded file.
+    dirname : string
+        The directory to save the file to.
+    file_checksum : string
+        The file checksum using sha256 to check against the one computed on the downloaded file.
+        Default is 'None', which means the checksum is not checked.
+    accept_license : boolean
+        Flag to specify if user accepts the file LICENSE and prevents from printing the corresponding license terms.
+        Default is False.
+
+    Returns
+    -------
+    file_path: string
+        Full path of the created file.
+
+    Raises
+    ------
+    IOError
+        If the computed SHA256 checksum of file does not match the one given by the user.
+    """
+
+    file_path = join(dirname, filename)
+
+    # Get the file
+    urlretrieve(url, file_path)
+
+    if file_checksum is not None:
+        checksum = _checksum_sha256(file_path)
+        if file_checksum != checksum:
+            # Remove file and raise error
+            remove(file_path)
+            raise IOError("{} has a SHA256 checksum : {}, "
+                        "different from expected : {}."
+                        "The file may be corrupted or the given url may be wrong !".format(file_path, checksum, file_checksum))
+
+    # Print license terms unless accept_license is set to True
+    if not accept_license:
+        license_file = join(dirname, "LICENSE")
+        if exists(license_file) and (file_path != license_file):
+            with open(license_file, 'r') as f:
+                print(f.read())
+
+    return file_path
+
+def _get_archive_and_dir(dirname, filename, label):
+    if dirname is None:
+        dirname = join(get_data_home(dirname), label)
+        makedirs(dirname, exist_ok=True)
+    else:
+        dirname = get_data_home(dirname)
+
+    archive_path = join(dirname, filename)
+
+    return archive_path, dirname
+
+def fetch_spiral_2d(filename = "spiral_2d.npy", dirname = None):
```

Comment:
(just thinking, not asking for some specific change)
I am trying to see if taking 2 arguments filename and dirname is a good idea. If I specify only filename, I get `~/gudhi_data/points/bunny/NAME`. I don't think there is much point specifying a non default name in such a location. If I specify only dirname, I get `bunny.npy` and `LICENSE` in `DIR`. If I fetch another dataset in the same `DIR`, its license file will overwrite the one for bunny. If I am specifying a directory, it isn't much more work to specify the full path (including the file name).
It looks like it would be simpler to ask for only `filename` (which can be `/tmp/rabbit.npy`, `./bun.npy`, etc) and default to `~/gudhi_data/points/bunny/bunny.npy`, although LICENSE files are inconvenient if we do not use subdirectories. We could avoid storing LICENSE if we only want to print it once, or go back to the `filename.LICENSE` format.

Reply:
Yes, specifying just the filename could be simpler... Another problem with the `LICENSE` file is that since it's stored, if I first fetch bunny in DIR and then fetch something else, spiral_2d for example, in the same DIR, the bunny LICENSE will be printed... So to avoid weird behaviors, we should either not store it or change the name with `filename` in it as you said...
Btw, why does `LICENSE` have this naming now? For simplicity?